### DISCUSSION TITLE
Confirmation bar of a dangerous action should move at a consistent speed

### DISCUSSION THREAD

**Discussion by monochrome22** [ppy/osu#22875] [Ideas]
The confirmation bar is fast at the start and slows down at the end. This can make it difficult to judge exactly how long you need to hold it, which can lead to a frustrating interaction. 
The most frustrating example is when I accidentally enter editor and have trouble leaving it. 

https://user-images.githubusercontent.com/122969285/225377848-23c83432-fec2-4e46-996b-a857011817e5.mp4

(I'm not intentionally moving the cursor away without completion, I'm moving it away because I think the bar will complete in time)

If the speed of the bar is made consistent, then the sound effect should be changed to match. 

Curious to hear if other people have the same experience.

**Comment by peppy** [comment]
This was a kinda intentional design choice so that when you let go of the mouse while almost filled, it wouldn't reset back as far.

Not sure if I'd want to change it.

**Reply by monochrome22** [reply]
By "design", do you mean visual design or UX design? 
If you mean UX design, then it doesn't make sense, because slowing down at the end doesn't change how long you need to hold the bar; the distance you need to travel is shorter, but if it's moving slower then you still need to hold it for the same amount of time.
If you mean visual design, then I would argue that the fancier visual appearance is not worth the potentially frustrating UX experience, especially for players who are new to lazer and are not used to the way the bar fills. Though even after a long time of using lazer, I still find the slowing confirmation bar to be frustrating to deal with. 

Since my post didn't get many upvotes, I would assume most people don't have an issue with the bar speed. But my argument for changing it would be: variable speed is fine for most, frustrating for some, while a consistent speed would be fine for everyone.
